analyses/new-feature-template.qmd

```diff
@@ -0,0 +1,1549 @@
+
+  ggplot(category_percentages, aes(x = !!sym(params$added_feature), y = percentage)) +
+    geom_bar(stat = "identity", fill = "blue", color = "black", alpha = 0.7) +
+    labs(
+      x = params$added_feature,
+      y = "Percentage"
+    ) +
+    theme_minimal()
+}
+```
+
+
+## FMV Change Histogram
+
+This chart shows the distribution of the value of 'diff_pred_pin_initial_fmv' in the model with the added feature minus the model without the added feature. Outliers outside of 95% are removed to make the chart more readable. The largest 100 increases and decreases are displayed in maps in section X. 
+```{r}
+working_data_pin %>%
+  filter(
+    diff_pred_pin_initial_fmv >= quantile(diff_pred_pin_initial_fmv, 0.025, na.rm = TRUE) &
+      diff_pred_pin_initial_fmv <= quantile(diff_pred_pin_initial_fmv, 0.975, na.rm = TRUE)
+  ) %>%
+  mutate(
+    mean_value = mean(diff_pred_pin_initial_fmv, na.rm = TRUE),
+    median_value = median(diff_pred_pin_initial_fmv, na.rm = TRUE)
+  ) %>%
+  ggplot(aes(x = diff_pred_pin_initial_fmv)) +
+  geom_histogram(fill = "blue", color = "black", alpha = 0.7) +
+  geom_vline(aes(xintercept = mean_value, color = "Mean"), linetype = "dashed", linewidth = 1, show.legend = TRUE) +
+  geom_vline(aes(xintercept = median_value, color = "Median"), linetype = "dashed", linewidth = 1, show.legend = TRUE) +
+  scale_color_manual(
+    name = "Statistics",
+    values = c(Mean = "red", Median = "green"),
+    labels = c("Mean", "Median")
+  ) +
+  labs(
+    x = "Change in FMV",
+    y = "Frequency"
+  ) +
+  theme_minimal()
+```
+
+## SHAP Histogram
+
+```{r}
+shap %>%
+  filter(
+    !!sym(params$added_feature) >= quantile(!!sym(params$added_feature), 0.025, na.rm = TRUE) &
+      !!sym(params$added_feature) <= quantile(!!sym(params$added_feature), 0.975, na.rm = TRUE)
+  ) %>%
+  mutate(
+    mean_value = mean(!!sym(params$added_feature), na.rm = TRUE),
+    median_value = median(!!sym(params$added_feature), na.rm = TRUE)
+  ) %>%
+  ggplot(aes(x = !!sym(params$added_feature))) +
+  geom_histogram(fill = "blue", color = "black", alpha = 0.7) +
+  geom_vline(aes(xintercept = mean_value, color = "Mean"), linetype = "dashed", linewidth = 1, show.legend = TRUE) +
+  geom_vline(aes(xintercept = median_value, color = "Median"), linetype = "dashed", linewidth = 1, show.legend = TRUE) +
+  scale_color_manual(
+    name = "Statistics",
+    values = c(Mean = "red", Median = "green"),
+    labels = c("Mean", "Median")
+  ) +
+  labs(
+    x = "SHAP Value",
+    y = "Frequency"
+  ) +
+  theme_minimal()
+```
+
+:::
+
+## Correlation between Added feature and Other features
+
+Here, the goal is to see if the added feature *very* neatly aligns with other existing features. Columns are produced with both the absolute value of the correlation (for easy sorting), as well as the correlation to help decipher the direction of the relationship.
+
+```{r correlation between features}
+clean_column_values <- function(df, column_name) {
```

Comment:
Do we have a dict for shortening variables. It seems really useful. Even this cleaning doesn't really do much for graphics.

Reply:
There is indeed a renaming function: [`ccao::vars_rename()`](https://ccao-data.github.io/ccao/reference/vars_rename.html). It probably needs its dictionary updated, but should work for the majority of features.